corehq/apps/userreports/tasks.py

```diff
@@ -294,13 +295,18 @@ def queue_async_indicators():
     start = datetime.utcnow()
     cutoff = start + ASYNC_INDICATOR_QUEUE_TIME - timedelta(seconds=30)
     retry_threshold = start - timedelta(hours=4)
-    # don't requeue anything that has been retried more than 20 times
-    indicators = AsyncIndicator.objects.filter(unsuccessful_attempts__lt=20)[:settings.ASYNC_INDICATORS_TO_QUEUE]
+    indicators = (
+        AsyncIndicator.objects
+        # don't requeue anything that has been retried more than 20 times
+        .filter(unsuccessful_attempts__lt=20)
+        # only requeue things that are not in queue or were last queued earlier than the threshold
```

Comment:
It looks like date_queued is really an indicator of whether an indicator is in queue or not. This is assumed by code, but is not made explicit. Might need a bit of public methods to make it more clear, but not going to consider in this PR.

Reply:
One of the downsides of this was that previously indicators contained a lot of objects that were already in queue so not many docs were getting queued effectively as ASYNC_INDICATORS_TO_QUEUE

Reply:
If there are 100 tasks queued a day, we would expect 100*ASYNC_INDICATORS_TO_QUEUE to be processed, but on datadog we can see this is not the case.

Reply:
to address all these comments
1) date_queued does indeed represent when it was queued for celery. It contrasts with date_created, which represents when it was added to the async_indicators table. I am not sure if there is clearer name, but it is certainly consistent and hopefully doesn't cause too much confusion
2) I am not sure this behavior is incorrect. This serves to regulate the number of queued indicators. The periodic task effectively brings the number of queued indicators to `ASYNC_INDICATORS_TO_QUEUE` each time it runs. If all previously queued indicators have been processed, it queues a full set. If only half have been processed, it queues half as many, but that would indicate that our processing is not keeping up with our queueing, not that we are failing to queue enough indicators.

Reply:
Okay, this was not at all obvious from the code, because the way this was being accomplished is by the fact that the AsyncIndicator has an implicit order-by on created_on, so [this line](https://github.com/dimagi/commcare-hq/pull/27207/files#diff-1f15a4ea1fec5c4ee5fa155d4d80c3d7L298) would return roughly same indicators all the time some of which might be in queue, so effectively queueing  not more than ASYNC_INDICATORS_TO_QUEUE.

This effectively means, we are never processing more than one indicator at a time and it doesn't matter how big or small ASYNC_INDICATORS_TO_QUEUE. Is that also intentional?